dev-docs/RFCs/v6.0/react-api.md

```diff
@@ -0,0 +1,151 @@
+>
+  <StaticMap
+    viewId="map"
+    {...this.state.viewState}
+    mapStyle={mapStyle}
+    mapboxApiAccessToken={mapboxApiAccessToken}
+  />
+</DeckGL>
+```
+
+- The `views` and `viewId` are required for `StaticMap` to receive dynamic `width` and `height` props.
+- `onViewStateChange` callback is required because `DeckGL` does not pass `viewState` to its children.
+- `viewId` is a non-standard prop and causes warnings when assigned to components such as `div`.
+- Regardless if a child component only desires to be placed at the correct offset in a multi-view canvas, its `width` and `height` would still be overwritten to the viewport size during render, which may cause render problems.
+
+
+### Usage of Deprecated React Lifecycle Methods
+
+`componentWillReceiveProps` is now marked [unsafe](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops) and will be removed in the next major version.
+
+## Proposal: Render Callbacks
+
+`DeckGL` accepts render callbacks as children. This is a common React pattern used by libraries such as [react-motion](https://github.com/chenglou/react-motion) and [react-virtualized](https://github.com/bvaughn/react-virtualized).
+
+- Static children are still supported.
+- Children in a single-view canvas no longer need to use `viewId` to subscribe to changes in the default view.
+- A child can determine how to best handle/discard the view information from the parent.
+- The user no longer have the burden to trigger rerender on viewport change, therefore able to leverage auto-control in a React app.
+
+**Stateless example:**
+
+```jsx
+<DeckGL
+  layers={layers}
+  viewState={this.state.viewState}
+  onViewStateChange={({viewState}) => this.setState({viewState})}
+  controller={MapController}
+>
+  {({width, height, viewState, viewport}) => <StaticMap
+    width={width}
+    height={height}
+    viewState={this.state.viewState}
+    mapStyle={mapStyle}
+    mapboxApiAccessToken={mapboxApiAccessToken}
+  />}
+</DeckGL>
+```
+
+**Statefull example:**
+
+```jsx
+<DeckGL
+  layers={layers}
+  initialViewState={INITIAL_VIEW_STATE}
+  onViewStateChange={console.log}
+  controller={MapController}
+>
+  {({width, height, viewState, viewport}) => <StaticMap
+    width={width}
+    height={height}
+    viewState={viewState}
+    mapStyle={mapStyle}
+    mapboxApiAccessToken={mapboxApiAccessToken}
+  />}
+</DeckGL>
+```
+
+**Using react-map-gl controls:**
+
+```jsx
+<DeckGL
+  layers={layers}
+  initialViewState={INITIAL_VIEW_STATE}
+  controller={MapController}
+>
+  {({width, height, viewState, viewport}) => labels.map(label => (
+    <Popup
+      key={label.id}
+      longitude={label.longitude}
+      latitude={label.latitude}
+      viewport={viewport}
+    >
+      {label.content}
+    </Popup>
+  ))}
+</DeckGL>
+```
+
+## Proposal: View Components
+
+Add `View` React component to the `@deck.gl/react` module. In the main `deck.gl` module, replace the `View` exports with their React counterparts.
```

Comment:
Would be good to avoid maintaining a growing set of react wrappers. It would be cool if the `View` element support could handle done "automatically" as an extension of the current JSX layers approach where we just check if a child `ReactElement` is a `View` in addition to checking if it is a `Layer`?

A possible drawback, is that while the API does looks pretty neat, it makes code that would otherwise be 95% portable between JS and React versions look quite different.

Reply:
That is a good idea. I will give it a try.